spec/sidekiq-instrument/server_middleware_spec.rb

```diff
@@ -40,53 +51,81 @@
-        let(:worker_metric_name) do
-          "sidekiq_instrument_trace_workers::in_queue"
-        end
-        it 'increments the enqueue counter' do
-            Sidekiq::Instrument::WorkerMetrics.enabled = true
-            Redis.new.hdel worker_metric_name ,'my_other_worker'
+      context 'with WorkerMetrics.enabled true and an errored job' do
+        it 'decrements the in queue counter' do
+          Sidekiq::Instrument::WorkerMetrics.enabled = true
+          MyOtherWorker.perform_async
+          expect(Redis.new.hget(worker_metric_name, 'my_other_worker')).to eq('-1')
+          begin
             MyOtherWorker.perform_async
-            expect(
-            Redis.new.hget worker_metric_name ,'my_other_worker'
-          ).to eq('-1')
+          rescue StandardError
+            nil
+          end
+          expect(Redis.new.hget(worker_metric_name, 'my_other_worker')).to eq('-2')
         end
       end
     end
 
     context 'when a job fails' do
-      before { allow_any_instance_of(MyWorker).to receive(:perform).and_raise('foo') }
+      before do
+        allow_any_instance_of(MyWorker).to receive(:perform).and_raise('foo')
+      end
 
       it 'increments the StatsD failure counter' do
-        expect {
-          MyWorker.perform_async rescue nil
-        }.to trigger_statsd_increment('shared.sidekiq.default.MyWorker.error')
+        expect do
+          MyWorker.perform_async
+        rescue StandardError
+          nil
+        end.to trigger_statsd_increment('shared.sidekiq.default.MyWorker.error')
       end
 
       it 'increments the DogStatsD failure counter' do
-        expect(Sidekiq::Instrument::Statter.dogstatsd).to receive(:increment).with('sidekiq.dequeue', expected_dog_options).once
+        expect(
+          Sidekiq::Instrument::Statter.dogstatsd
+        ).to receive(:increment).with('sidekiq.dequeue', expected_dog_options).once
         expect(Sidekiq::Instrument::Statter.dogstatsd).not_to receive(:time)
-        expect(Sidekiq::Instrument::Statter.dogstatsd).to receive(:increment).with('sidekiq.error', expected_dog_options).once
-        MyWorker.perform_async rescue nil
+        expect(
+          Sidekiq::Instrument::Statter.dogstatsd
+        ).to receive(:increment).with('sidekiq.error', expected_dog_options).once
+
+        begin
+          MyWorker.perform_async
+        rescue StandardError
+          nil
+        end
+      end
+
+      it 'does not increase the redis counter' do
+        expect(Redis.new.hget(worker_metric_name, 'my_worker')).to eq(nil)
+        begin
+          MyWorker.perform_async
+        rescue StandardError
+          nil
+        end
       end
 
       it 're-raises the error' do
         expect { MyWorker.perform_async }.to raise_error 'foo'
       end
+
+      it 'calls the decrement counter' do
+        expect(
+          Sidekiq::Instrument::WorkerMetrics
+          ).to receive(:trace_workers_decrement_counter).with('my_worker').once
```

Comment:
Nit (Rubocop flags this, too): bump the paren back one level to match its expect( indentation
```suggestion
        ).to receive(:trace_workers_decrement_counter).with('my_worker').once
```